npm/ipsm/ipsm_test.go

```diff
@@ -294,45 +283,42 @@ func TestDeleteSet(t *testing.T) {
 
 func TestAddToSet(t *testing.T) {
 	metrics.NumIPSetEntries.Set(0)
-	ipsMgr := NewIpsetManager(exec.New())
-	if err := ipsMgr.Save(util.IpsetTestConfigFile); err != nil {
-		t.Fatalf("TestAddToSet failed @ ipsMgr.Save")
-	}
-
-	defer func() {
-		if err := ipsMgr.Restore(util.IpsetTestConfigFile); err != nil {
-			t.Fatalf("TestAddToSet failed @ ipsMgr.Restore")
-		}
-	}()
 
 	testSetName := "test-set"
-	if err := ipsMgr.AddToSet(testSetName, "1.2.3.4", util.IpsetNetHashFlag, ""); err != nil {
-		t.Fatalf("TestAddToSet failed @ ipsMgr.AddToSet")
+	var calls = []testutils.TestCmd{
+		{Cmd: []string{"ipset", "-N", "-exist", util.GetHashedName(testSetName), "nethash"}},
+		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.2.3.4"}},
+		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.2.3.4/", "nomatch"}},
+		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1,tcp:8080"}},
+		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1,:"}}, // todo: verify this is proper formatting
+		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1"}},
 	}
 
-	if err := ipsMgr.AddToSet(testSetName, "1.2.3.4/nomatch", util.IpsetNetHashFlag, ""); err != nil {
-		t.Fatalf("TestAddToSet with nomatch failed @ ipsMgr.AddToSet %v", err)
-	}
+	fexec := testutils.GetFakeExecWithScripts(calls)
+	ipsMgr := NewIpsetManager(fexec)
+	defer testingutils.VerifyCalls(t, fexec, calls)
+
+	err := ipsMgr.AddToSet(testSetName, "1.2.3.4", util.IpsetNetHashFlag, "")
```

Comment:
Can we leverage table driven tests to maintain and understand testsets better?
For example, we use below codes to have the same tests. Since there are no dependencies between calls, I think below logics will be easier to maintain testsets later (e.g., adding more testsets) and understand testsets as well.
```go
	testSetName := "test-set"
	var calls = []testutils.TestCmd{
		{Cmd: []string{"ipset", "-N", "-exist", util.GetHashedName(testSetName), "nethash"}},
		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.2.3.4"}},
		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.2.3.4/", "nomatch"}},
		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1,:"}}, // todo: verify this is proper formatting
		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1"}},
		{Cmd: []string{"ipset", "-A", "-exist", util.GetHashedName(testSetName), "1.1.1.1,tcp:8080"}},
	}

	fexec := testutils.GetFakeExecWithScripts(calls)
	ipsMgr := NewIpsetManager(fexec)
	defer testingutils.VerifyCalls(t, fexec, calls)

	ipsets := [][]string{
		{testSetName, "1.2.3.4", util.IpsetNetHashFlag, ""},
		{testSetName, "1.2.3.4/nomatch", util.IpsetNetHashFlag, ""},
		{testSetName, fmt.Sprintf("%s,:", "1.1.1.1"), util.IpsetIPPortHashFlag, "0"},
		{testSetName, fmt.Sprintf("%s", "1.1.1.1"), util.IpsetIPPortHashFlag, "0"},
		{testSetName, fmt.Sprintf("%s,%s:%d", "1.1.1.1", "tcp", 8080), util.IpsetIPPortHashFlag, "0"},
	}

	for _, ipset := range ipsets {
		err := ipsMgr.AddToSet(ipset[0], ipset[1], ipset[2], ipset[3])
		require.NoError(t, err)
	}

	ipsets = [][]string{
		{testSetName, fmt.Sprintf("%s,%s:%d", "", "tcp", 8080), util.IpsetIPPortHashFlag, "0"},
		{testSetName, fmt.Sprintf(""), util.IpsetIPPortHashFlag, "0"},
	}

	for _, ipset := range ipsets {
		err := ipsMgr.AddToSet(ipset[0], ipset[1], ipset[2], ipset[3])
		require.Error(t, err)
	}
```

If it looks reasonable, would you please walk through ipsm_test.go to take same approaches?
I think there are multiple places where we can leverage table driven tests (e.g., TestCreateSet, TestDeleteFromList, etc).

In addition, in all testsets, "test-list" and "test-set" are used. So, I think it is ok to declare them as global variables in this file  and use them. So, when we want to change them to other values, we can change only one line.